Simple-Projects/Simple_Projects/Basic/List/List.cpp

```diff
@@ -28,28 +27,23 @@ int List::max_size() const {
 }
 
 void List::erase(int index) {
-    if ( current > index && index > 0 ) {
-        current -= 1;
-
-        for (int i = 0; i < current; i++ ) {
-        	array[i] = array[i+1];
-        }
+    if ( current > index && index >= 0 ) {
+		std::copy(array + index + 1, array + current, array);
+        --current;
     }
 }
 
 void List::insert(int value, int index) {
     if ( current >= index && index >= 0 ) {
-        push_back(value);
+        push_back(0);
 
-        for ( int i = current; i > index; i-- ) {
-            array[i] = array[i-1];
-        }
+		std::copy_backward(array + index, array + current - 1, array + current);
         array[index] = value;
     }
 }
 
 int List::find(int value) const {
-    for ( int i = 0 ; i < current; i++ ){
+    for ( int i = 0 ; i < current; ++i ) {
```

Comment:
Operators ++ and -- behave differently when used in prefix and postfix forms.
When you only need to alter the variable and need not retain the old value in a temporary variable, prefer the prefix form.
The most noticeable difference is for iterators with overloaded operators.